### ISSUE TITLE
New Rule Proposal: No multiple assignments per line

### ISSUE THREAD

**Issue by stewx** [eslint/eslint#7862]
**Please describe what the rule should do:**
It should disallow writing lines like `a = b = c = d;`

**What category of rule is this?** (place an "X" next to just one item)
[X] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**
Good:
```js
var a = 1,
    b = 2,
    c = 3,
    d = 4;
a = d;
b = d;
c = d;
```

Bad:
```js
var a = 1,
    b = 2,
    c = 3,
    d = 4;
a = b = c = d; // Bad line
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Assigning variable values in this way may be unclear to certain readers.

**Comment by not-an-aardvark** [comment]
This looks like a duplicate of https://github.com/eslint/eslint/issues/6424.

The other issue is already accepted, so if you're interested in implementing this rule, we would be happy to look at a PR. :smile:

**Comment by stewx** [comment]
Created PR here https://github.com/eslint/eslint/pull/7904
